### ISSUE TITLE
Use more than one project

### ISSUE THREAD

**Issue by TimMeade** [n6g7/redux-saga-firebase#98]
Can redux-saga-firebase use more than one project?  We are using the auth and firestore sections and it's working well with a single project.   

We are trying to open a second project and are giving the init a second name..  seems to start, but the data is still the to other project.  Very odd.

admin.initializeApp(config, 'secondProject');

Has anyone had success with this?

**Comment by n6g7** [comment]
Hi @TimMeade,

I've never used several firebase apps simultaneously but [the docs](https://firebase.google.com/docs/web/setup#initialize_multiple_apps) suggest that this should be possible with something like the following:

```js
// Initialize both firebase apps + rsf instances
const app1 = firebase.initializeApp(app1Config, 'project1')
const rsf1 = new ReduxSagaFirebase(app1)

const app1 = firebase.initializeApp(app2Config, 'project2')
const rsf2 = new ReduxSagaFirebase(app2)

function * saga () {
  // 👇Login the user using project1
  const user = yield call(
    rsf1.auth.loginWithPopup,
    provider
  )
  // 👇Save data in project2
  yield call(
    rsf2.firestore. setDocument,
    'collection/document',
    { user }
  )
}
```

It'd be easier to help you if you could provide a code example of something that doesn't work. 🙂

**Comment by n6g7** [comment]
I looked a bit more into it and I think the problem could be coming from the way firestore is initialized.

Our current doc suggests doing the following:
```js
const reduxSagaFirebase = new ReduxSagaFirebase(myFirebaseApp, firebase.firestore())
```
But I suspect that this might not work when `myFirebaseApp` isn't the default app because `firebase.firestore()` is actually a shortcut for the default app (it's equivalent to `defaultApp.firestore()`). So we might be happening is that we're linking a firestore instance to the wrong app.

So this might fix your issue:
```js
const reduxSagaFirebase = new ReduxSagaFirebase(myFirebaseApp, myFirebaseApp.firestore())
```
Notice we're doing `myFirebaseApp.firestore()` instead of `firebase.firestore()`.

I tested this locally with a single app and it works as expected. Please let me know if solves your issue and I'll update the docs.

**Comment by TimMeade** [comment]
Good news everyone!

Yes this works.  I moved some stuff around to what we had working.  Here is what I have now for the SECOND project.   The first default project is the standard one.

reduxSagaFirebaseTeleco.js is the loading file.
```
import firebase from 'firebase';
import 'firebase/firestore';
import ReduxSagaFirebase from 'redux-saga-firebase';

const config = {
    {...config stuff from firestore}
};

const telecoFirebase = firebase.initializeApp(config, 'teleco');

const reduxSagaFirebaseTeleco = new ReduxSagaFirebase(telecoFirebase, telecoFirebase.firestore());
const firestoreTeleco = telecoFirebase.firestore();
export { reduxSagaFirebaseTeleco, firestoreTeleco };
```

saga.js (watches and returns changes to calls collection)
```
import { all, call, takeLatest } from 'redux-saga/effects';
import { syncCalls } from './actions';
import CALLS from './constants';
import { reduxSagaFirebaseTeleco, firestoreTeleco } from './reduxSagaFirebaseTeleco';

function* getCalls() {
  const callsTransformer = (calls) => {
    const res = [];
    calls.forEach(doc => res.push({
      id: doc.id,
      ...doc.data(),
    }));
    return res;
  };

  yield call(
    reduxSagaFirebaseTeleco.firestore.syncCollection,
    firestoreTeleco.collection('calls')
      .where('callStatus', '==', 'OPEN')
      .orderBy('callStart'),
    {
      successActionCreator: syncCalls,
      transform: callsTransformer,
    },
  );
}

export default function* rootSaga() {
  yield all([
    takeLatest(CALLS.GET, getCalls),
  ]);
}
```

Much thanks for the quick help....

**Comment by n6g7** [comment]
Great!

This issue also made me realize that, since we have access to `firebaseApp.firestore()` instead of `firebase.firestore()`, we actually don't need to pass a second firestore argument to the `ReduxSagaFirebase` constructor to be able to use firestore.

I opened [a PR](https://github.com/n6g7/redux-saga-firebase/pull/99) which takes advantage of this. Fortunately, the API change is backward compatible (we would just ignore the second argument). I'd love your feedback on this. :)

**Comment by n6g7** [comment]
Hi Tim,

I released [v0.12.0](https://github.com/n6g7/redux-saga-firebase/releases/tag/v0.12.0) with the changes described.
You should be able to update without any breaking changes and then you can remove the second argument to the RSF constructor (the firestore instance). Hopefully, that means no more headaches!

I'll close this issue for now but please let me know if this problem comes back. :)
